mito-ai/src/components/DropdownMenu.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState, useRef, useEffect } from 'react';
+import '../../style/DropdownMenu.css';
+
+interface DropdownMenuItem {
+    label: string;
+    onClick: () => void;
+    icon?: React.ComponentType<{ fill?: string }>;
+    iconColor?: string; // This can be a hex color or a variable
+}
+
+interface DropdownMenuProps {
+    trigger: React.ReactNode;
+    items: DropdownMenuItem[];
+    className?: string;
+}
+
+const DropdownMenu: React.FC<DropdownMenuProps> = ({ trigger, items, className = '' }) => {
+    const [isOpen, setIsOpen] = useState(false);
+    const dropdownRef = useRef<HTMLDivElement>(null);
+
+    // Close dropdown when clicking outside
+    useEffect(() => {
+        const handleClickOutside = (event: MouseEvent) => {
+            if (dropdownRef.current && !dropdownRef.current.contains(event.target as Node)) {
+                setIsOpen(false);
+            }
+        };
+
+        document.addEventListener('mousedown', handleClickOutside);
+        return () => document.removeEventListener('mousedown', handleClickOutside);
+    }, []);
+
+    const handleItemClick = (onClick: () => void) => {
+        onClick();
+        setIsOpen(false);
+    };
+
+    return (
+        <div className={`dropdown-container ${className}`} ref={dropdownRef}>
+            <div onClick={() => setIsOpen(!isOpen)}>
+                {trigger}
+            </div>
+            {isOpen && (
+                <div className="dropdown-menu">
+                    {items.map((item, index) => (
+                        <button
+                            key={index}
+                            className={`dropdown-item ${item.icon ? 'dropdown-item-with-icon' : ''}`}
+                            onClick={() => handleItemClick(item.onClick)}
+                            style={{ display: 'flex', gap: '5px', alignItems: 'center' }}
+                        >
+                            {item.icon && (
```

Comment:
Let's make sure that the dropdown item icon container always takes up a consistent width. Like: 20px or whatever regardless of the size of the icon inside. 

Right now the text is not aligned. 
<img width="160" alt="Screenshot 2025-02-07 at 1 06 05 PM" src="https://github.com/user-attachments/assets/231dfdda-e2e5-4753-b6c2-3ddb903c495d" />